server/src/main/java/org/elasticsearch/search/aggregations/metrics/WrapperTDigestArrays.java

```diff
@@ -0,0 +1,365 @@
+        // have to if the delta is negative
+        if (delta > 0) {
+            try {
+                breaker.addEstimateBytesAndMaybeBreak(delta, "tdigest-array-wrapper");
+            } catch (CircuitBreakingException e) {
+                // since we've already created the data, we need to
+                // add it so closing the stream re-adjusts properly
+                breaker.addWithoutBreaking(delta);
+                throw e;
+            }
+        } else {
+            breaker.addWithoutBreaking(delta);
+        }
+    }
+
+    private abstract static class MemoryAwareArray implements Releasable, Accountable {
+        protected final CircuitBreaker breaker;
+        private final AtomicBoolean closed = new AtomicBoolean(false);
+
+        MemoryAwareArray(CircuitBreaker breaker) {
+            this.breaker = breaker;
+        }
+
+        @Override
+        public final void close() {
+            if (closed.compareAndSet(false, true)) {
+                breaker.addWithoutBreaking(-ramBytesUsed());
+            }
+        }
+    }
+
+    public static class WrapperTDigestDoubleArray extends MemoryAwareArray implements TDigestDoubleArray {
+        static final long SHALLOW_SIZE = RamUsageEstimator.shallowSizeOfInstance(WrapperTDigestDoubleArray.class);
+
+        private double[] array;
+        private int size;
+
+        public WrapperTDigestDoubleArray(CircuitBreaker breaker, int initialSize) {
+            this(breaker, new double[initialSize]);
+        }
+
+        public WrapperTDigestDoubleArray(CircuitBreaker breaker, double[] array) {
+            super(breaker);
+            this.array = array;
+            this.size = array.length;
+        }
+
+        @Override
+        public long ramBytesUsed() {
+            return SHALLOW_SIZE + RamUsageEstimator.sizeOf(array);
+        }
+
+        @Override
+        public int size() {
+            return size;
+        }
+
+        @Override
+        public double get(int index) {
+            assert index >= 0 && index < size;
+            return array[index];
+        }
+
+        @Override
+        public void set(int index, double value) {
+            assert index >= 0 && index < size;
+            array[index] = value;
+        }
+
+        @Override
+        public void add(double value) {
+            ensureCapacity(size + 1);
+            array[size++] = value;
+        }
+
+        @Override
+        public void sort() {
+            Arrays.sort(array, 0, size);
+        }
+
+        @Override
+        public void resize(int newSize) {
+            ensureCapacity(newSize);
+
+            if (newSize > size) {
+                Arrays.fill(array, size, newSize, 0);
+            }
+
+            size = newSize;
+        }
+
+        @Override
+        public void ensureCapacity(int requiredCapacity) {
+            if (requiredCapacity > array.length) {
+                int newSize = array.length + (array.length >> 1);
```

Comment:
Nit: is this 50% increase? Why not double? Please add a comment too.

Reply:
50% yep. Just added a comment. The original code used `ArrayList`, which from my investigation, did this to grow. So I tried to keep the same behaviour

Reply:
Just updated: Changed to `ArrayUtil.oversize()` (As with grow I can't call the CB before array creation)